### ISSUE TITLE
dependency of a dependency does not deploy in correct order

### ISSUE THREAD

**Issue by chasebolt** [devspace-cloud/devspace#946]
**What happened?**  
Dependencies failed to deploy in order I expected. Project A depends on B which depends on C. Project B was deployed before C which failed the deployment.

**What did you expect to happen instead?**  
Project A depends on B which depends on C. I expected C to be deploy, then B, and then finally A.

**How can we reproduce the bug?** (as minimally and precisely as possible)  
Project B (showing C as a dependency): https://gist.github.com/chasebolt/44aafa1397803b007b1315c8eeba03e6
Project C: https://gist.github.com/chasebolt/9098d9b33b92ba4ffc4b2ddee47b380c

**Local Environment:**  
- Operating System: linux
- Deployment method: helm

**Kubernetes Cluster:**  
- Cloud Provider: minikube
- Kubernetes Version:
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-11T18:14:22Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.9", GitCommit:"2e808b7cb054ee242b68e62455323aa783991f03", GitTreeState:"clean", BuildDate:"2020-01-18T23:24:23Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```

**Anything else we need to know?**  

/kind bug

**Comment by FabianKramm** [comment]
@chasebolt thanks for opening this issue! Yes you are correct, this is a bug, currently the order is reversed, I'll create a fix today.

**Comment by FabianKramm** [comment]
Should be fixed with #948

**Comment by chasebolt** [comment]
thank you
